### PR TITLE
[WIP] Split manual and recurring backups

### PR DESCRIPTION
**Pull request by moelsayed**
https://github.com/rancher/rancher/issues/18807
https://github.com/rancher/rancher/issues/18793

### CODE REVIEW

File: pkg/controllers/management/etcdbackup/etcdbackup.go [hunk 1]
```diff
@@ -250,14 +244,17 @@ func (c *Controller) rotateExpiredBackups(cluster *v3.Cluster, clusterBackups []
 	internvalHours := cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.IntervalHours
 	expiredBackups := getExpiredBackups(retention, internvalHours, clusterBackups)
 	for _, backup := range expiredBackups {
+		if backup.Spec.Manual {
```
Comment (alena1108) [inline pkg/controllers/management/etcdbackup/etcdbackup.go]: can you confirm that this code will result in manually taking snapshots not being removed, for both local and and the ones backed up to S3

File: pkg/controllers/management/etcdbackup/etcdbackup.go [hunk 2]
```diff
@@ -332,6 +330,20 @@ func (c *Controller) deleteS3Snapshot(b *v3.EtcdBackup) error {
 	return s3Client.RemoveObject(bucket, b.Name)
 }
 
+func (c *Controller) getRecuringBackupsList(cluster *v3.Cluster) ([]*v3.EtcdBackup, error) {
+	retList := []*v3.EtcdBackup{}
```
Comment (alena1108) [inline pkg/controllers/management/etcdbackup/etcdbackup.go]: nit: slice is better be initialized like `var retList []*v3.EtcdBackup`

File: pkg/controllers/management/etcdbackup/etcdbackup.go [hunk 3]
```diff
@@ -364,21 +376,28 @@ func shouldBackup(cluster *v3.Cluster) bool {
...
-	if etcdService.BackupConfig != nil &&
-		!*cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.Enabled {
-		logrus.Debugf("[etcd-backup] Backup is disabled cluster [%s]", cluster.Name)
+	// we only work with ready clusters
+	if !v3.ClusterConditionReady.IsTrue(cluster) {
+		return false
+	}
+
+	if !isRecurringBackupEnabled(cluster.Spec.RancherKubernetesEngineConfig) {
+		logrus.Debugf("[etcd-backup] Recurring backup is disabled cluster [%s]", cluster.Name)
```
Comment (alena1108) [inline pkg/controllers/management/etcdbackup/etcdbackup.go]: nit: *for* a cluster in the log

**Comment by kinarashah** [comment]
Addressed review comments, and rebased -  https://github.com/rancher/rancher/pull/19623

File: pkg/controllers/management/etcdbackup/etcdbackup.go (hunk 1 above)
Comment (kinarashah) [reply pkg/controllers/management/etcdbackup/etcdbackup.go]: Ya, all backup configs have `manual` irrespective of provider flag, https://github.com/rancher/rancher/blob/6ff16315e29ff7a1ace33413a19a4232617d7493/pkg/controllers/management/etcdbackup/etcdbackup.go#L285

**Comment by kinarashah** [comment]
Pulled in changes into #19623.

**Pull request closed without merging**